include/qubes-gui-protocol.h

```diff
@@ -99,7 +99,19 @@ enum {
     MSG_WMCLASS,
     MSG_WINDOW_DUMP,
     MSG_CURSOR,
-    MSG_MAX
+    MSG_MAX,
+    MSG_KEYPRESS2,
+    MSG_BUTTON2,
+    MSG_MOTION2,
+};
+enum {
+    FLAG_KEYPRESS_REPEAT=0b1,
+};
+enum {
+    FLAG_BUTTON_EMULATED=0b1,
+};
+enum {
+    FLAG_MOTION_EMULATED=0b1,
```

Comment:
Are those flags actually useful for an application in the VM? It may be seen as a feature that VM has no idea whether event was emulated or not

Reply:
Yes. Mouse wheel events are mapped as emulated button presses. Some games rely on motion events, and get insane mouse wheel speed when they can't filter these out.

Reply:
Maybe we don't send repeated or emulated events to client VMs, and let the client generate those events themselves? This way, each client can have different key repeat intervals.